Revocation Api (EXPOSUREAPP-12554)

BMItr opened the pull request:
This PR addresses Exposureapp-[12554](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12554).
Revocation API. (Subtask)

BMItr committed:
Created RevocationParser

BMItr committed:
Added RevocationParserTest

BMItr committed:
added explanations to EntryCoord types

BMItr committed:
Merge branch 'feature/12747-revocation-parser' into feature/12554-revocation-api

BMItr committed:
Implemented RevocationApi

BMItr committed:
Implemented RevocationServer

BMItr committed:
Shorter param name

BMItr committed:
- Implemented API creation, Added RevocationCache

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/storage/RevocationRepository.kt:
```diff
@@ -22,6 +25,11 @@ class RevocationRepository @Inject constructor(
             scope = appScope
         )
+
+    fun clear() {
```
Please add to DataResetModule? in order not to forget it

BMItr replied:
good point!
adjusted.

mtwalli committed:
Merge branch 'feature/12455-feature-Revocation-of-single-certs' into feature/12554-revocation-api

BMItr committed:
Merge branch 'feature/12455-feature-Revocation-of-single-certs' into feature/12554-revocation-api

BMItr committed:
Adjusted RevocationServer

BMItr committed:
Added RevocationRepository to DataReset

BMItr committed:
Added RevocationServerTest

BMItr committed:
Update HashExtensionsTest.kt

BMItr committed:
clean

mtwalli committed:
Merge branch 'feature/12455-feature-Revocation-of-single-certs' into feature/12554-revocation-api

mtwalli committed:
Pay technical debt 💸

https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-app-android&branch=feature%2F12555-revocation-storage-repo&resolved=false&types=CODE_SMELL

mtwalli reviewed (approved):
LGTM 🖖🏼
